### ISSUE TITLE
Honor JS properties manually set on components before initialization

### ISSUE THREAD

**Issue by LeaVerou** [shoelace-style/webawesome#650]
I was reminded of this by https://github.com/shoelace-style/webawesome-alpha/discussions/141#discussioncomment-12031207

The fact that JS properties set on components before initialization are not honored is something that complicates any JS example using WA components, as its author needs to understand esoteric Lit details like `updateComplete`, making it a [leaky abstraction](https://en.wikipedia.org/wiki/Leaky_abstraction).

I would even go as far as to say this is a bug — components are not working as they should, and do not behave like regular elements.

This is not a platform limitation — it’s a Lit one. It's entirely possible to honor these properties, you just need to check whether they are already set upon element initialization, which presumably already happens since otherwise they would shadow the accessors defined on the prototype.

Ideally, we should report this to Lit (though there must already be a report I bet) and we could explore ways to work around it in our own components.

**Comment by claviska** [comment]
Even if this could be fixed, it won't work for methods since the element isn't upgraded yet. While Lit's abstraction does leak a bit, so does the platforms in that loading a `<script type="module">` will defer loading, making this break, which is unintuitive.

```html
<script type="module" src="webawesome.js"></script>

<script>
  const input = document.querySelector('wa-input');

  // What users expect to do
  input.select(); // Error: the component isn't registered yet

  // What users need to do
  await customElements.whenDefined('wa-select');
  input.select();
</script>
```

So they need to have _some_ kind of awareness of this when they're not using frameworks, because that's how the platform works.

BTW — I'm confident that user's problems will be solved by using the npm repo. It's always been a pain to load custom elements in Vue (and other frameworks) without registering them upfront, which Vite will do when imported from npm instead of the CDN.

**Comment by LeaVerou** [comment]
True, but I think people hit this *a lot* more with properties than methods, and the fact their values are not retained seems a lot more surprising for properties — whereas for methods, it's easier to understand that if the element is not yet upgraded, the method won't be there.

> BTW — I'm confident that user's problems will be solved by using the npm repo. It's always been a pain to load custom elements in Vue (and other frameworks) without registering them upfront, which Vite will do when imported from npm instead of the CDN.

This will solve the issues for Vue, but what I'm suggesting will solve (most of) them for any workflow. They're not mutually exclusive though, we can (and should) do both!

**Comment by LeaVerou** [comment]
I was reminded of this by [this customer issue](https://github.com/shoelace-style/webawesome-alpha/discussions/209), which I suspect might be rooted in this.

I talked with Justin Fagnani a few weeks ago and he said they are copying over these properties and this shouldn't happen, the properties should be retained. So it's not a Lit issue, it must be something WA is doing. He asked for a reduced testcase so he can investigate.

**Comment by LeaVerou** [comment]
So after a bit more testing, it seems to be specific components, while others honor these properties just fine. So I'm gonna close this and open specific issues for the specific components when I encounter this.
